### ISSUE TITLE
[ally]: Combobox Default & with Layer - VO focus option on open dropdown

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon-website#3554]
## Environment

> Operating system

> Browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?
After double tapping on the "editcombo", user should be able to swipe right and reach the options. Note: However there is an accessible alternative is available here. the user can still activate the "button" present after the combobox and access the same options. Hence this issue is mentioned only as an observation.

When the focus is not set on a logical point in the page after a user action, it creates a great deal of confusion. The whole process of navigating back to the relevant element can be tedious. note this defect mainly impacts keyboard-only users and those with cognitive disabilities. 

> What happened instead?
In Vo iPhone, After double tapping the "edit combo" when we swipe Right to reach the options, the options automatically get collapsed. Here the user has to activate the "open pop-up" button to access the options.

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

Default: https://react.carbondesignsystem.com/?path=/story/components-combobox--default
With layer: https://react.carbondesignsystem.com/?path=/story/components-combobox--with-layer

## Additional information

- Screenshots or code
- Notes
